### PR TITLE
Solve formatting issues (stylish-haskell, pre-commit CI)

### PR DESCRIPTION
**Pull request by andys8**
Because of the failing pre-commit hook https://github.com/haskell/haskell-language-server/issues/3170 a bunch of files are checked in, but not formatted as expected.

Note: The commit hook will still fail because of the first reason (stylish haskell parse errors, https://github.com/haskell/haskell-language-server/issues/3170)

Solves formatting issues

image

Solves some of the parse errors Currently results in ...

image

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/3171"></a>

### CODE REVIEW

**Comment by kokobd** [comment]
Have you checked #3131 and #3145 out?

**Comment by andys8** [comment]
@kokobd I haven't. So thanks for the hint.

* #3131 appears to be closed. Is the work gone or in a different PR?
* #3145 seems to address running the hook only on changed files. Would not be relevant.

In case this PR is duplicated, I'm happy to close it ;)

**Comment by drsooch** [comment]
The closed PR I eventually found a file that needed a complete overhaul to remove the stylish Haskell error.

#3145 is intended to run just changes files to hopefully minimize the amount of formatting failures on CI. I have an approval but am going to try to remove the dependency on another action when I get some spare time.

File: ghcide/src/Development/IDE/Core/Shake.hs [hunk 1]
```diff
@@ -130,11 +130,8 @@ import           Development.IDE.GHC.Compat             (NameCache,
                                                          NameCacheUpdater (..),
                                                          initNameCache,
                                                          knownKeyNames,
-#if !MIN_VERSION_ghc(9,3,0)
```
Comment (andys8) [inline ghcide/src/Development/IDE/Core/Shake.hs:133]: :warning: Need to take a look why this condition was dropped

Comment (andys8) [reply ghcide/src/Development/IDE/Core/Shake.hs:133]: Converting to draft because of this

Comment (andys8) [reply ghcide/src/Development/IDE/Core/Shake.hs:133]: https://github.com/haskell/haskell-language-server/pull/3171/commits/e134351387792feb35936940da188b0210866588 merged, resolves some of the issues

**Comment by andys8** [comment]
e134351387792feb35936940da188b0210866588 addresses some of the issues, but not all

**Review by pepeiborra — APPROVED** [review]
I don't see any downsides to merging this. Thanks!

**Pull request merged**